src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +39,113 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
+    return (stop - start + step - 1) / step;
+}
+
+class StridedSliceShapeInfer : public ShapeInferEmptyPads {
+public:
+    StridedSliceShapeInfer(const std::shared_ptr<ov::Node>& op,
+                           const IShapeInfer::port_mask_t& port_mask,
+                           const std::set<int64_t>& begin_mask,
+                           const std::set<int64_t>& end_mask,
+                           const std::set<int64_t>& new_axis_mask,
+                           const std::set<int64_t>& shrink_axis_mask)
+    : m_op(op), m_port_mask(port_mask), m_outputShape(op->get_output_partial_shape(0).rank().get_length(), 1),
+      m_begin_mask_set(begin_mask), m_end_mask_set(end_mask), m_new_axis_mask_set(new_axis_mask), m_shrink_axis_mask_set(shrink_axis_mask) {}
+
+    std::vector<VectorDims> infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& shapeIn = input_shapes[0].get();
+        const VectorDims& shapeBegin = input_shapes[1].get();
+        auto beginPtr = reinterpret_cast<int *>(data_dependency.at(1)->GetPtr());
+        auto endPtr = reinterpret_cast<int *>(data_dependency.at(2)->GetPtr());
+        auto stridePtr = reinterpret_cast<int *>(data_dependency.at(3)->GetPtr());
+
+        for (auto i=0, new_idx=0; i<shapeIn.size(); ++i) {
+            bool is_new_axis = m_new_axis_mask_set.count(i);
+            bool is_shrink_axis = m_shrink_axis_mask_set.count(i);
+            bool is_begin_axis = m_begin_mask_set.count(i);
+            bool is_end_axis = m_end_mask_set.count(i);
+
+            if (is_new_axis) {
+                m_outputShape[new_idx] = 1;
+                m_outputShape[new_idx+1] = shapeIn[i];
+                new_idx+=2;
+                continue;
+            }
+            if (is_shrink_axis) {
+                continue;
+            }
+            if (i<shapeBegin[0]) {
+                auto begin = is_begin_axis? 0 : beginPtr[i];
+                auto end  = is_end_axis? shapeIn[i] : endPtr[i];
+                m_outputShape[new_idx] = calculate_output_shape(begin, end, stridePtr[i], shapeIn[i]);
+            } else {
+                m_outputShape[new_idx] = shapeIn[i];
+            }
+            new_idx += 1;
+        }
+        return {m_outputShape};
+    }
+
+    port_mask_t get_port_mask() const override {
+        return m_port_mask;
+    }
+
+private:
+    const std::shared_ptr<ov::Node> m_op;
+    const IShapeInfer::port_mask_t m_port_mask;
+    VectorDims m_outputShape;
+    // set
+    std::set<int64_t> m_begin_mask_set;
+    std::set<int64_t> m_end_mask_set;
+    std::set<int64_t> m_new_axis_mask_set;
+    std::set<int64_t> m_shrink_axis_mask_set;
+};
+
+class StridedSliceShapeInferFactory : public ShapeInferFactory {
+public:
+    StridedSliceShapeInferFactory(const std::shared_ptr<ov::Node>& op, const IShapeInfer::port_mask_t& port_mask)
```

Comment:
```suggestion
    StridedSliceShapeInferFactory(const std::shared_ptr<ov::Node>& op, IShapeInfer::port_mask_t port_mask)
```
Maybe I'm missing something, but why it is better to pass 32bit value by reference?

Reply:
> Maybe I'm missing something, but why it is better to pass 32bit value by reference?

reply as above. accept your advice.